extensions/data-plane-selector/data-plane-selector-control-api/src/test/java/org/eclipse/edc/connector/dataplane/selector/control/api/DataplaneSelectorControlApiControllerTest.java

```diff
@@ -139,6 +143,156 @@ void shouldFail_whenEgressTransformationFails() {
+
+            given()
+                    .port(port)
+                    .contentType(JSON)
+                    .body(Json.createObjectBuilder().build())
+                    .post("/v1/dataplanes/select")
+                    .then()
+                    .statusCode(404);
+        }
+
+        @Test
+        void shouldReturnInternalServerError_whenEgressTransformationFails() {
+            var sourceAddress = DataAddress.Builder.newInstance().type("sourceType").build();
+            var selectionRequest = SelectionRequest.Builder.newInstance()
+                    .source(sourceAddress)
+                    .transferType("transferType")
+                    .strategy("strategy")
+                    .build();
+            when(typeTransformerRegistry.transform(any(), eq(SelectionRequest.class)))
+                    .thenReturn(Result.success(selectionRequest));
+            var dataPlane = DataPlaneInstance.Builder.newInstance()
+                    .url("http://any-url")
+                    .build();
+            when(service.select(any(), anyString(), anyString()))
+                    .thenReturn(ServiceResult.success(dataPlane));
+            when(typeTransformerRegistry.transform(any(), eq(JsonObject.class)))
+                    .thenReturn(Result.failure("error"));
+
+            given()
+                    .port(port)
+                    .contentType(JSON)
+                    .body(Json.createObjectBuilder().build())
+                    .post("/v1/dataplanes/select")
+                    .then()
+                    .statusCode(500);
+        }
+    }
+
+    @Nested
+    class GetAll {
+
+        @Test
+        void shouldReturnAllDataplaneInstances() {
+            var dataPlane = DataPlaneInstance.Builder.newInstance()
+                    .url("http://any-url")
+                    .build();
+            when(service.getAll()).thenReturn(ServiceResult.success(List.of(dataPlane)));
+            when(typeTransformerRegistry.transform(any(), eq(JsonObject.class)))
+                    .thenReturn(Result.success(Json.createObjectBuilder().build()));
+
+            given()
+                    .port(port)
+                    .get("/v1/dataplanes")
+                    .then()
+                    .statusCode(200)
+                    .contentType(JSON)
+                    .body("size()", is(1));
+        }
+
+        @Test
+        void shouldIgnoreFailedTransformations() {
+            var dataPlane = DataPlaneInstance.Builder.newInstance()
+                    .url("http://any-url")
+                    .build();
+            when(service.getAll()).thenReturn(ServiceResult.success(List.of(dataPlane)));
+            when(typeTransformerRegistry.transform(any(), eq(JsonObject.class)))
+                    .thenReturn(Result.failure("error"));
+
+            given()
+                    .port(port)
+                    .get("/v1/dataplanes")
+                    .then()
+                    .statusCode(200)
```

Comment:
in other APIs we have this as a 500, for example an egress transformation error on selection (Line 218)

Reply:
yep, that's right. I kinda lean towards always throwing an exception, but tackling this globally in a subsequent cleanup issue/pr makes sense.